### PR TITLE
feat: Add sample to listen error pubsub

### PR DESCRIPTION
**Pull request by hemanshv**
Add pubsub sample to listen to error while pulling messages.

### CODE REVIEW

**Comment by hemanshv** [comment]
Hi @amanda-tarafa ,
This is under our internal review now. We will let you know once this is done.

Thanks,

File: pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+// [START pubsub_subscriber_error_listener]
+
+using Google.Cloud.PubSub.V1;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class PullMessagesAsyncWithErrorListen
```
Comment (Rishabh-V) [inline pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs:23]: Based on existing sample pattern, I think the class should be named `PullMessagesWithErrorListenerAsyncSample` and file should be named `PullMessagesWithErrorListenerAsync.cs`

File: pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+{
+    public async Task<int> PullMessagesAsyncListenError(string projectId, string subscriptionId, bool acknowledge)
```
Comment (Rishabh-V) [inline pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs:25]: The method name should be `PullMessagesWithErrorListenerAsync`

File: pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        // SubscriberClient runs your message handle function on multiple
+        // threads to maximize throughput.
+        int messageCount = 0;
+        Task startTask = subscriber.StartAsync((PubsubMessage message, CancellationToken cancel) =>
+        {
+            string text = System.Text.Encoding.UTF8.GetString(message.Data.ToArray());
+            Console.WriteLine($"Message {message.MessageId}: {text}");
+            Interlocked.Increment(ref messageCount);
+            return Task.FromResult(acknowledge ? SubscriberClient.Reply.Ack : SubscriberClient.Reply.Nack);
+        }).ContinueWith(task =>
```
Comment (Rishabh-V) [inline pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs:38]: I am not sure what we want to demonstrate as part of this sample. But corresponding Java sample says - Listen for unrecoverable failures. Rebuild a subscriber and restart subscribing when the current subscriber encounters permanent errors. We are not doing this. Can we please check and confirm what is it that we want to demonstrate here?

File: pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            Console.WriteLine("Exception while listening msgs: ", task.Exception.InnerException.Message);
+        }, TaskContinuationOptions.OnlyOnFaulted);
+
+        // Run for 5 seconds.
+        await Task.Delay(5000);
+        await subscriber.StopAsync(CancellationToken.None).ContinueWith(task =>
```
Comment (Rishabh-V) [inline pubsub/api/Pubsub.Samples/PullMessagesAsyncWithErrorListen.cs:45]: Why do we need it?

**Review by Rishabh-V — COMMENTED** [review]
I had a quick look at the sample. Can we please check and confirm what we want to demonstrate in this sample?

**Comment by amanda-tarafa** [comment]
@hemanshv Is this PR still relevant?

**Pull request closed without merging**